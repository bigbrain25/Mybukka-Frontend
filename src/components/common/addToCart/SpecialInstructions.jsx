/* eslint-disable react/prop-types */
import React, { useState } from 'react';
import PropTypes from 'prop-types';

import { connect } from 'react-redux';

import TextArea from 'Components/input/TextArea';
import manipulateMealAction from 'Redux/manipulateMealAction';

import SelectQuantityButtons from './SelectQuantityButtons';

import './specialInstructions.scss';

const TextLengthSection = ({ length }) => (
  <div className="text-length-section">
    <h5 className="text-length-indicator">{length}/200</h5>
  </div>
);

const SpecialInstructions = ({
  handleChange, itemIsInCart, quantity, manipulateMeal
}) => {
  const [textLength, setTextLength] = useState(0);

  const onChangeHandler = ({ target: { value } }) => {
    handleChange(value);
    setTextLength(value);
  };

  return (
    <div className="special-instructions-section">
      <div className="title-special-instruction-section">
        <h4 className="title-special-instruction">SPECIAL INSTRUCTIONS</h4>
      </div>
      <TextArea
        type="textarea"
        name="specialInstructions"
        maxLength={200}
        handleChange={onChangeHandler}
        handleFocus={() => {}}
      />
      <TextLengthSection length={textLength} />
      <div className="d-lg-none quantity-toggler-md">
        <SelectQuantityButtons
          itemIsInCart={itemIsInCart}
          quantity={quantity}
          manipulateMeal={manipulateMeal}
        />
      </div>
    </div>
  );
};

export default connect(() => ({}), {
  manipulateMeal: manipulateMealAction
})(SpecialInstructions);

TextLengthSection.defaultProps = {
  length: 0
};

TextLengthSection.propTypes = {
  length: PropTypes.number
};
