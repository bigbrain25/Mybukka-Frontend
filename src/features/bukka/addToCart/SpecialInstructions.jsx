import React, { useState } from 'react';
import PropTypes from 'prop-types';

import TextArea from 'Components/input/TextArea';

import SelectQuantityButtons from './SelectQuantityButtons';

import './specialInstructions.scss';

const TextLengthSection = ({ length }) => (
  <div className="text-length-section">
    <h5 className="text-length-indicator">{length}/200</h5>
  </div>
);

const SpecialInstructions = () => {
  const [textLength, setTextLength] = useState(0);

  return (
    <div className="special-instructions-section">
      <div className="title-special-instruction-section">
        <h4 className="title-special-instruction">SPECIAL INSTRUCTIONS</h4>
      </div>
      <TextArea
        type="textarea"
        name="specialInstructions"
        maxLength={200}
        handleChange={event => setTextLength(event.target.value.length)}
        handleFocus={() => {}}
      />
      <TextLengthSection length={textLength} />
      <div className="d-lg-none quantity-toggler-md">
        <SelectQuantityButtons />
      </div>
    </div>
  );
};

export default SpecialInstructions;

TextLengthSection.defaultProps = {
  length: 0
};

TextLengthSection.propTypes = {
  length: PropTypes.number
};
