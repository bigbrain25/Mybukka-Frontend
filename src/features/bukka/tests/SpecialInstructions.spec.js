import React from 'react';
import SpecialInstructions from '../addToCart/SpecialInstructions';

describe('Special Instructions Component', () => {
  const wrapper = mount(<SpecialInstructions />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it(`changes the text length indicator value
  as the user types in textarea`, () => {
    const textArea = wrapper.find('textarea');
    textArea.simulate('change', { target: { value: 'a new instruction' } });

    const textLengthIndicatorValue = wrapper
      .find('.text-length-indicator')
      .props().children;

    expect(textLengthIndicatorValue[0]).toEqual(17);
  });
});
