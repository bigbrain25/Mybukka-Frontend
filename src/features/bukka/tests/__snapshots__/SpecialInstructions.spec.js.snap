// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Special Instructions Component renders properly 1`] = `ReactWrapper {}`;
